src/main/resources/data/computercraft/lua/rom/startup.lua

```diff
@@ -81,9 +81,17 @@ shell.setCompletionFunction("rom/programs/monitor.lua", completion.build(
         if previous[2] == "scale" then
             return completion.peripheral(shell, text, previous, true)
         else
-            return completion.program(shell, text, previous)
+            return completion.programWithArgs(shell, text, previous, 3)
         end
-    end
+    end,
+    {
+        function(shell, text, previous)
```

Comment:
Yikes, OK. This is ugly D:. Maybe we need a `buildv2` function which does things right :tm:

Reply:
There is a risk it would end up getting overbuild for its uses.